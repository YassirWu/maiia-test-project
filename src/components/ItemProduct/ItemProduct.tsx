import * as React from 'react';
import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';
import Product from 'model/Product';
import { makeStyles } from '@material-ui/styles';

const useStyle = makeStyles({
  root: {
    marginBottom: '10px',
  },
  title: {
    fontWeight: 'normal',
  },
  add: {
    borderRadius: '20px'
  }
})

export interface IItemProductProps {
  product: Product;
  onAddProductToShoppingCart: () => void;
}

const ItemProduct: React.FunctionComponent<IItemProductProps> = ({ product, onAddProductToShoppingCart }) => {
  const classes = useStyle();
  
  return (
    <Card className={classes.root}>
      <Card.Img src={product.thumbnailUrl} />
      <Card.Body>
        <Card.Title className={classes.title}><p>{product.title}</p>
          <Button onClick={onAddProductToShoppingCart} variant="danger" className={classes.add}>+</Button>
        </Card.Title>
      </Card.Body>
    </Card>
  );
};

export default ItemProduct;
