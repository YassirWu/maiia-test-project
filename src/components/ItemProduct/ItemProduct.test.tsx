import React from 'react';
import { shallow } from 'enzyme';
import ItemProduct from './ItemProduct';
import Product from 'model/Product';

describe('<ItemProduct />', () => {
  it('should render ItemProduct component', () => {
    const product: Product = {
      id: 1,
      albumId: 2,
      title: 'my product',
      thumbnailUrl: 'thumb url',
      url: 'url',
    };

    const wrapper = shallow(
      <ItemProduct product={product} onAddProduct={jest.fn()} />
    );

    expect(wrapper).toMatchSnapshot();
  });
})


