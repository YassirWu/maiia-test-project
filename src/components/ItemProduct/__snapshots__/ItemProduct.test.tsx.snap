// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemProduct /> should render ItemProduct component 1`] = `ShallowWrapper {}`;
